smack-experimental/src/main/java/org/jivesoftware/smackx/mam/MamManager.java

```diff
@@ -214,6 +215,33 @@ public MamQueryResult queryArchiveWithEndDate(Date end) throws NoResponseExcepti
     public MamQueryResult queryArchive(Integer max, Date start, Date end, Jid withJid, List<FormField> additionalFields)
             throws NoResponseException, XMPPErrorException, NotConnectedException, InterruptedException,
             NotLoggedInException {
+      return queryArchive(null, null, max, start, end, withJid, additionalFields);
+    }
+
+
+    /**
+     * Query an archive object, lke a room or a pub-sub node, applying filters: max count, start date, end date, from/to
+     * JID and with additional fields.
+     * 
+     * When objectJID is null the default, server will be requested.
+     * 
+     * @param objectJID The object for which archive is requested
+     * @param node The node name, can be null
+     * @param max
+     * @param start
+     * @param end
+     * @param withJid
+     * @param additionalFields
+     * @return the MAM query result
+     * @throws NoResponseException
+     * @throws XMPPErrorException
+     * @throws NotConnectedException
+     * @throws InterruptedException
+     * @throws NotLoggedInException
+     */
+    public MamQueryResult queryArchive(Jid objectJID, String node, Integer max, Date start, Date end, Jid withJid, List<FormField> additionalFields)
```

Comment:
Please change `objectJid` to `mamArchiveAddress`. (BTW: How did you come up with "objectJid"?)